### PR TITLE
Update release workflow to include new docker images

### PR DESCRIPTION
**Pull request by adchia**
**What this PR does / why we need it**:
There are new feature transformation server, lambda feature server, and java feature server docker images. This adds convenience commands in the Makefile + merges/updates our release workflow to build + push these images to GCR/dockerhub.

This also removes a workflow that builds a new FTS image + tags against the commit. We don't use this (the helm charts don't point to these either).

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -129,34 +129,50 @@ lint-go:
...
+		-f sdk/python/feast/infra/transformation_servers/Dockerfile .
 
-push-serving-docker:
-	docker push $(REGISTRY)/feast-serving:$(VERSION)
-
-build-core-docker:
-	docker build --build-arg VERSION=$(VERSION) -t $(REGISTRY)/feast-core:$(VERSION) -f infra/docker/core/Dockerfile .
+push-feature-server-java-docker:
+	docker push $(REGISTRY)/feature-server-java:$(VERSION)
+    docker push gcr.io/kf-feast/feature-server-java:$(VERSION)
```
Comment (pyalex) [inline Makefile:168]: A little while ago we migrated to docker hub as a default container registry? Are we moving back to gcr.io?

**Review by pyalex — APPROVED** [review]

File: .github/workflows/master_only.yml [hunk 2]
```diff
@@ -134,3 +134,43 @@ jobs:
         run: pytest --verbose --color=yes sdk/python/tests --integration --benchmark --benchmark-autosave --benchmark-save-data --durations=5
       - name: Upload Benchmark Artifact to S3
         run: aws s3 cp --recursive .benchmarks s3://feast-ci-pytest-benchmarks
+
+  build-publish-docker-images:
```
Comment (tsotnet) [inline .github/workflows/master_only.yml:138]: Why do we need to publish these images for every commit?

Comment (adchia) [reply .github/workflows/master_only.yml:138]: Removed as discussed offline. Keeping the build though for a continuous build

**Review by tsotnet — APPROVED** [review]

**Pull request merged**